src/server/services/wms/qgswmsrestorer.cpp

```diff
@@ -94,13 +84,9 @@ QgsLayerRestorer::~QgsLayerRestorer()
     // if a SLD file has been loaded for rendering, we restore the previous style
     if ( layer->customProperty( "readSLD", false ).toBool() )
     {
-      QString errMsg;
-      QDomElement root = settings.mQgisStyle.documentElement();
-      QgsReadWriteContext context = QgsReadWriteContext();
-      if ( !layer->readStyle( root, errMsg, context ) )
-      {
-        QgsMessageLog::logMessage( QStringLiteral( "QGIS Style has not been read from layer restorer for layer %1: %2" ).arg( layer->name(), errMsg ) );
-      }
+      QString sldStyleName = layer->customProperty( "sldStyleName" ).toString();
```

Comment:
@rldhont What do you think about using only one custom property? Because in this case I think we can remove the `readSLD` bool property and directly check `sldStyleName` with something like:

``` cpp
const QString prop = layer->customProperty( "sldStyleName", "" ).toString();
if ( ! prop.isEmpty() )
{
    ...
}
```

Reply:
Thanks for the review. I will propose some code to remove `readSLD`.

Reply:
@pblottiere done